packages/embark-transaction-manager/src/lib/eth_accounts.ts

```diff
@@ -0,0 +1,61 @@
+import { Callback, Embark, Events, Logger } /* supplied by @types/embark in packages/embark-typings */ from "embark";
+import Web3 from "web3";
+const { blockchain: blockchainConstants } = require("embark-core/constants");
+import { __ } from "embark-i18n";
+import { AccountParser, dappPath } from "embark-utils";
+import { EventEmitter } from "events";
+import Transaction from "./Transaction";
+
+const TXS_TO_MODIFY = [
+  blockchainConstants.transactionMethods.eth_accounts,
+  blockchainConstants.transactionMethods.personal_listAccounts,
+];
+
+function arrayEqual(arrayA: string[], arrayB: string[]) {
+  if (arrayA.length !== arrayB.length) {
+    return false;
+  } else {
+    return arrayA.every((address, index) => Web3.utils.toChecksumAddress(address) === Web3.utils.toChecksumAddress(arrayB[index]));
+  }
+}
+
+export default class EthAccounts implements Transaction {
+  public readonly events: Events;
+  public readonly logger: Logger;
+  constructor(readonly embark: Embark, readonly web3: Web3, public nodeAccounts: any[], public accounts: any[], readonly txEvents: EventEmitter) {
+    this.events = embark.events;
+    this.logger = embark.logger;
+
+    if (accounts.length) {
+      this.embark.registerActionForEvent("blockchain:proxy:response", this.checkResponseFor_eth_accounts.bind(this));
+    }
+  }
+
+  private async checkResponseFor_eth_accounts(params: any, callback: Callback<any>) {
+
+    if (!(TXS_TO_MODIFY.includes(params.reqData.method))) {
```

Comment:
Jut realized, `TXS_TO_MODIFY` is not really the right word, it should be more `METHODS_TO_MODIFY`, because `eth_accounts` and the other are not txs.

Reply:
Updated.